packages/connect-node/src/grpc-web-protocol.ts

```diff
@@ -65,9 +68,14 @@ export function createGrpcWebProtocol(
       switch (spec.kind) {
         case MethodKind.Unary:
           return async (req, res) => {
+            const requestHeader = nodeHeaderToWebHeader(req.headers);
             const type = grpcWebParseContentType(
               req.headers["content-type"] ?? null
             );
+            const timeout = parseGrpcTimeoutFromHeader(
```

Comment:
If this fails to parse, we are raising an error, which is only going to be caught in the fallback [here](https://github.com/bufbuild/connect-web/blob/6ca3d00ca3f59f735df9b014427049559220c83d/packages/connect-node/src/handler.ts#L123). 

Instead, we need to return this error to the client. So for this protocol, you need to call `endWithGrpcWebTrailer()` with the error that occurred during parsing of `grpc-timeout`.

Please parse the timeout _after_ checking whether the content-type is valid. We should not try to parse the `grpc-timeout` header if we don't recognize the `content-type` header.

I think this is passing the tests because `timeout_on_sleeping_server` also goes green for `Code.Unknown`, which is a relic from testing against envoy. Can you remove `Code.Unknown` from timeout_on_sleeping_server.spec.ts:51 to confirm?

Reply:
removed the check for `Code.Unknown` and test still pass